build.xml

```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" DefaultTargets="Default">
+    <!-- TODO use common build lifecycle -->
+    <Import Project="build\common.props" />
+
+    <!-- build project dependencies -->
+    <PropertyGroup>
+        <RestoreSources>$(RestoreSources);https://dotnet.myget.org/F/aspnetcore-tools/api/v3/index.json</RestoreSources>
+        <!-- this build project targets netcoreapp1.0 so it can restore build targets, but doesn't actually compile sources -->
+        <TargetFramework>netcoreapp1.0</TargetFramework>
+    </PropertyGroup>
+    <ItemGroup>
+        <PackageReference Include="NuGetPackageVerifier" Version="1.0.1-*" />
+    </ItemGroup>
+
+    <!-- build settings -->
+    <PropertyGroup>
+        <ArtifactsPath>$(MSBuildThisFileDirectory)artifacts\build\</ArtifactsPath>
+        <SolutionFile>$(MSBuildThisFileDirectory)DotNetTools.sln</SolutionFile>
+    </PropertyGroup>
+    <ItemGroup>
+        <SourceProjects Include="src\*\*.csproj" />
+        <TestProjects Include="test\*\*.csproj" />
+    </ItemGroup>
+
+    <Target Name="Initialize">
+        <Message Importance="High" Text="Build settings:" />
+        <Message Importance="High" Text="DotnetCliVersion: $(DotnetCliVersion)" />
+        <MSBuild Projects="$(SolutionFile)" Targets="Restore" />
+    </Target>
+
+    <Target Name="Clean">
+        <RemoveDir Directories="$(ArtifactsPath);$(MSBuildThisFileDirectory)testWorkDir" />
+        <MSBuild Projects="$(SolutionFile)" Targets="Clean" />
+    </Target>
+
+    <Target Name="Pack">
+        <MSBuild Targets="Rebuild;Pack"
+                 Projects="@(SourceProjects)"
+                 Properties="PackageOutputPath=$(ArtifactsPath)" />
+    </Target>
+
+    <Target Name="Test">
+        <MSBuild Targets="VSTest"
+                 Projects="@(TestProjects)" />
```

Comment:
👍 Would be good if it works well. I'll play it. Not sure how well VSTest handles parallel test projects.